Content.Server/Explosion/Components/TwoStagedGrenadeComponent.cs

```diff
@@ -0,0 +1,50 @@
+using Content.Server.Singularity.Components;
+using Content.Shared.Singularity.Components;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Explosion.Components;
+
+/// <summary>
+/// This component after trigger starts timer to trigger for second time to explode or delete self.
+/// This component is controlling process of exploding of supermatter grenade.
+/// </summary>
+[RegisterComponent]
+public sealed class TwoStagedGrenadeComponent : Component
```

Comment:
Grenades shouldn't be hardcoded to be supermatter or the likes, it should be compositional.

Reply:
I still don't know way how to add OnTrigger components in List of this component to just 
```suggestion
/// <summary>
/// This component after trigger starts timer to trigger for second time to explode or delete self.
/// For example this is controlling process of exploding of supermatter grenade.
/// </summary>
[RegisterComponent]
public sealed class TwoStagedGrenadeComponent : Component
```